bin/peggy.mjs

```diff
@@ -0,0 +1,341 @@
+  }
+  return f;
+}
+
+// Command line processing
+
+const program = new Command();
+let options = program
+  .version(peg.VERSION, "-v, --version")
+  .arguments("[input_file]")
+  .addOption(
+    new Option(
+      "--allowed-start-rules <rules>",
+      "comma-separated list of rules the generated parser will be allowed to start parsing from"
+    )
+      .default([], "the first rule in the grammar")
+      .argParser((val, prev) => prev.concat(val.split(",").map(x => x.trim())))
+  )
+  .option(
+    "--cache",
+    "make generated parser cache results"
+  )
+  .option(
+    "-d, --dependency <dependency>",
+    "use specified dependency (can be specified multiple times)",
+    (val, prev) => (prev || []).concat([val])
+  )
+  .option(
+    "-e, --export-var <variable>",
+    "name of a global variable into which the parser object is assigned to when no module loader is detected"
+  )
+  .option(
+    "--extra-options <options>",
+    "additional options (in JSON format as an object) to pass to peg.generate",
+    (val, prev) => addExtraOptions(prev, val)
+  )
+  .option(
+    "-c, --extra-options-file <file>",
+    "file with additional options (in JSON as an object or commonjs module format) to pass to peg.generate",
+    (val, prev) => {
+      if (/\.c?js$/.test(val)) {
+        return Object.assign({}, prev, require(path.resolve(val)));
+      } else {
+        return addExtraOptions(prev, readFile(val));
+      }
+    }
+  )
+  .addOption(
+    new Option(
+      "--format <format>",
+      "format of the generated parser"
+    )
+      .choices(MODULE_FORMATS)
+      .default(DEFAULT_FORMAT, '"commonjs"')
+  )
+  .option("-o, --output <file>", "output file")
+  .option(
+    "--plugin <module>",
+    "use a specified plugin (can be specified multiple times)",
+    (val, prev) => (prev || []).concat([val])
+  )
+  .option(
+    "-t, --test <text>",
+    "Test the parser with the given text, outputting the result of running the parser instead of the parser itself"
+  )
+  .option(
+    "-T, --test-file <filename>",
+    "Test the parser with the contents of the given file, outputting the result of running the parser instead of the parser itself"
+  )
+  .option("--trace", "enable tracing in generated parser")
+  .addOption(
+    // Not interesting yet.  If it becomes so, unhide the help.
+    new Option("--verbose", "enable verbose logging")
+      .hideHelp()
+      .default(false)
+  )
+  .addOption(
+    new Option("-O, --optimize <style>")
+      .hideHelp()
+      .argParser(() => {
+        console.error("Option --optimize is deprecated from 1.2.0 and has no effect anymore.");
+        console.error("It will be deleted in 2.0.");
+        console.error("Parser will be generated in the former \"speed\" mode.");
+        return "speed";
+      })
+  )
+  .parse()
+  .opts();
+
+if (options.extraOptions) {
+  Object.assign(options, options.extraOptions);
+  delete options.extraOptions;
```

Comment:
I see potential problem with that deletion. If custom options define the `extraOptions` key, it will be deleted here. Need to delete it before doing
```js
Object.assign(options, options.extraOptions);
```